packages/tabs/src/LionTabs.js

```diff
@@ -0,0 +1,248 @@
+      return;
+    }
+    this.__store.forEach(entry => {
+      this.__cleanButton(entry.button, entry.clickHandler, entry.keydownHandler);
+    });
+  }
+
+  __setupPanel({ element, uid }) {
+    element.setAttribute('id', `panel-${uid}`);
+    element.setAttribute('role', 'tabpanel');
+    element.setAttribute('aria-labelledby', `button-${uid}`);
+    this.__deselectPanel(element);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  __selectPanel(element) {
+    element.setAttribute('selected', true);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  __deselectPanel(element) {
+    element.removeAttribute('selected');
+  }
+
+  __setupButton({ element, uid, clickHandler, keydownHandler }) {
+    element.setAttribute('id', `button-${uid}`);
+    element.setAttribute('role', 'tab');
+    element.setAttribute('aria-controls', `panel-${uid}`);
+    element.addEventListener('click', clickHandler);
+    element.addEventListener('keyup', keydownHandler);
+    this.__deselectButton(element);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  __cleanButton(element, clickHandler, keydownHandler) {
+    element.removeAttribute('id');
+    element.removeAttribute('role');
+    element.removeAttribute('aria-controls');
+    element.removeEventListener('click', clickHandler);
+    element.removeEventListener('keyup', keydownHandler);
+  }
+
+  __createButtonClickHandler(index) {
+    return () => {
+      this.selectedIndex = index;
+    };
+  }
+
+  __handleButtonKeydown(e) {
+    switch (e.key) {
+      case 'ArrowDown':
+      case 'ArrowRight':
+        e.preventDefault();
+        if (this.selectedIndex + 1 >= this._pairCount) {
+          this.selectedIndex = 0;
+        } else {
+          this.selectedIndex += 1;
+        }
+        break;
+      case 'ArrowUp':
+      case 'ArrowLeft':
+        e.preventDefault();
+        if (this.selectedIndex <= 0) {
+          this.selectedIndex = this._pairCount - 1;
+        } else {
+          this.selectedIndex -= 1;
+        }
+        break;
+      case 'Home':
+        e.preventDefault();
+        this.selectedIndex = 0;
+        break;
+      case 'End':
+        e.preventDefault();
+        this.selectedIndex = this._pairCount - 1;
+        break;
+      default:
+      // nothing
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  __selectButton(element) {
+    element.focus();
+    element.setAttribute('selected', true);
+    element.setAttribute('aria-selected', true);
+    element.setAttribute('tabindex', 0);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  __deselectButton(element) {
+    element.removeAttribute('selected');
+    element.setAttribute('aria-selected', false);
+    element.setAttribute('tabindex', -1);
+  }
+
+  set selectedIndex(value) {
+    const stale = this.__selectedIndex;
+    this.__selectedIndex = value;
+    this.__updateSelected();
+    // TODO: discuss before or after request update?!
```

Comment:
doesn't matter in this case as for setter/getters the value is already updated by the time you fire the event (which is what we want)

`this.requestUpdate` is async so if anyone needs it to be done they need to await updateComplete